packages/flutter/lib/src/material/button_bar.dart

```diff
@@ -172,3 +181,182 @@ class ButtonBar extends StatelessWidget {
     return null;
   }
 }
+
+/// Attempts to display buttons in a row, but displays them in a column if
+/// there is not enough horizontal space.
+///
+/// It first attempts to lay out its buttons as though there were no
+/// maximumm width constraints on the widget. If the button bar's width is
+/// less than the maximum width constraints of the widget, it then lays
+/// out the widget as though it were placed in a [Row].
+///
+/// However, if the button bar's width exceeds the maximum width constraint on
+/// the widget, it then aligns its buttons in a column. The key difference here
+/// is that the [MainAxisAlignment] will then be treated as a
+/// cross-axis/horizontal alignment. For example, if the buttons overflow and
+/// [ButtonBar.alignment] was set to [MainAxisAligment.start], the buttons would
+/// align to the horizontal start of the button bar.
+class ButtonBarRow extends Flex {
+  /// Creates a button bar that attempts to display in a row, but displays in
+  /// a column if there is insufficient horizontal space.
+  ButtonBarRow({
```

Comment:
Do we really want this as a public class? Do we envision apps using this directly? Seems like this is just an implementation detail. It might be confusing to have both ButtonBar` and `ButtonBarRow` as part of the public API.

Reply:
Good point. I made this private. We can always make this public if the need arises